jdaviz/core/template_mixin.py

```diff
@@ -281,24 +281,42 @@ def hub(self):
     def viewer_dicts(self):
         def _dict_from_viewer(viewer, viewer_item):
             d = {'viewer': viewer,
-                 'id': viewer_item['id'],
-                 'icon': self.app.state.viewer_icons.get(viewer_item['id'])}
+                 'id': viewer_item.get('id'),
+                 'icon': self.app.state.viewer_icons.get(viewer_item.get('id'))}
             if viewer_item.get('reference') is not None:
-                d['reference'] = viewer_item['reference']
-                d['label'] = viewer_item['reference']
+                d['reference'] = viewer_item.get('reference')
+                d['label'] = viewer_item.get('reference')
             else:
                 d['reference'] = None
-                d['label'] = viewer_item['id']
+                d['label'] = viewer_item.get('id')
```

Comment:
No problems here, just curious; is this a stylistic decision or a code-based decision? As far as I know, `get` with no fallback behaves the same as `[]`

Reply:
get with no fallback fallsback to a fallback of `None`

Reply:
Ahh got it (I think... 😵). Thanks!